Sources/GRPCHTTP2TransportNIOPosix/HTTP2ClientTransport+Posix.swift

```diff
@@ -133,11 +140,37 @@ extension HTTP2ClientTransport.Posix {
     func establishConnection(
       to address: GRPCHTTP2Core.SocketAddress
     ) async throws -> HTTP2Connection {
+      #if canImport(NIOSSL)
+      let nioSSLContext: NIOSSLContext?
+      switch self.config.transportSecurity.wrapped {
+      case .plaintext:
+        nioSSLContext = nil
+      case .tls(let tlsConfig):
+        do {
+          nioSSLContext = try NIOSSLContext(configuration: TLSConfiguration(tlsConfig))
+        } catch {
+          throw RuntimeError(
+            code: .transportError,
+            message: "Couldn't create SSL context, check your TLS configuration.",
+            cause: error
+          )
+        }
+      }
+      #endif
+
       let (channel, multiplexer) = try await ClientBootstrap(
         group: self.eventLoopGroup
       ).connect(to: address) { channel in
         channel.eventLoop.makeCompletedFuture {
-          try channel.pipeline.syncOperations.configureGRPCClientPipeline(
+          #if canImport(NIOSSL)
+          if let nioSSLContext {
+            try channel.pipeline.syncOperations.addHandler(
+              NIOSSLServerHandler(context: nioSSLContext)
```

Comment:
copy-pasta? Should be `NIOSSLClientHandler`

Reply:
Ah, yes 🫠 
Now, the client requires an (optional) hostname to validate certificates against. But we only have the server's address, not its hostname. Should we get the hostname as part of the TLS config?

Reply:
Ignore that comment - just saw your comment about the TLSConfig answering my question :D